src/access/acp.test.ts

```diff
@@ -235,250 +256,105 @@ function mockResourceWithAcr(
   const plainResource = mockSolidDatasetFrom(accessTo);
   return addMockAcrTo(plainResource, acr);
 }
+function mockAcpData(
+  mockPolicies: Partial<MockPolicies> = {}
+): internal_AcpData {
+  const rulesByUrl: Record<UrlString, Rule> = {};
+  function generateMockRule(mockRuleUrl: UrlString, mockRule: MockRule) {
+    let rule = rulesByUrl[mockRuleUrl] ?? createRule(mockRuleUrl);
+    Object.entries(mockRule).forEach(([mockActorRelation, mockActors]) => {
+      mockActors?.forEach((mockActor) => {
+        rule = addIri(rule, mockActorRelation, mockActor);
+      });
+    });
+    rulesByUrl[mockRuleUrl] = rule;
+    return rule;
+  }
```

Comment:
I did indeed try something like that at first, but since the scope is still limited to the containing function and it was getting quite involved, I figured I'd just keep it like this.

The refactoring does make sense (actually, I think `mockAcr` could probably just reuse `mockAcpData` directly) - it wasn't obvious while working on it that they would end up so similar. I'll take a look at that.

Reply:
OK, so I looked into this, but the challenge here is that `mockAcpData` does not generate Member (ACR) Policies, because those aren't used by the universal access API's. However, the existing tests do include tests that verify that if a Member Policy is defined by not a regular Policy, that the Universal Access API's do not apply that Policy.

So in other words, to support that `mockAcr` needs to generate Member Policies as well, which `mockAcpData` does not do. That means there are three possible approaches:

1. Update `AcpData` to also include Member Policies (but this would result in additional HTTP requests that are not used, but could potentially fail).
2. Remove the tests that check that Member Policies do not affect the outcome.
3. Leave things as-is.

I wasn't too enthusiastic about the first two, so I went with the third. We can always choose to do 1. or 2. in a future PR if need be :)